artiq/gateware/shuttler.py

```diff
@@ -113,8 +118,14 @@ def __init__(self):
             Dds(self.clear),
         ]
 
+        # Infer signed multiplication
+        data_raw = Signal((14, True))
+        data_buf = Signal(14)
         self.sync.rio += [
-            self.data.eq(reduce(add, [sub.data for sub in subs])),
+            data_raw.eq(reduce(add, [sub.data for sub in subs])),
+            # Extra buffer for better DSP timing
+            data_buf.eq(((data_raw * Cat(self.gain, ~self.gain[-1])) + (self.offset << 16))[16:]),
```

Comment:
`data` can overflow/underflow here when `gain` > 1. Should I detect and handle these case (probably by setting `data` to ceiling or floor)?